carrot-store-UI/details/details.js

```diff
@@ -0,0 +1,88 @@
+function getDataFromServer(productId) {
+    fetch("http://localhost:3000/products/" + productId)
+        .then(displayResponse)
+        .then(displayData);
+}
+
+
+// logic to get de id from query param
+function getParameterByName(name, url) {
+    if (!url) url = window.location.href;
+    name = name.replace(/[\[\]]/g, '\\$&');
+    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+        results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
+    return decodeURIComponent(results[2].replace(/\+/g, ' '));
+}
+
+
+console.log("query param: " + window.location.search)
+
+
+productId = getParameterByName("id", window.location.search);
+console.log("productId is " + productId);
+
+if (productId === null) {
+    productId = "0"
+} 
+
```

Comment:
It is possible to assign a default value to productId by using the || (OR) operator, e.g.

**productId = getParameterByName("id", window.location.search) || "0"**

This would filter out any "falsy" value ("", 0, null, undefined, etc...)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators#Logical_OR_2